test/testdata/lsp/completion/locals.rb

```diff
@@ -0,0 +1,92 @@
+# typed: true
+extend T::Sig
+
+# This is not defined in any of the methods below.
+# It should never show up in a completion result inside a method.
+x_outside = nil
+
+x_ # error: does not exist
+# ^ completion: x_outside
+
+def without_args
+  x_1 = nil
+  x_ # error: does not exist
+  # ^ completion: x_1
+end
+
+def with_args(x_1)
+  x_2 = nil
+  x_ # error: does not exist
+  # ^ completion: x_1, x_2
+end
+
+# We sort the local variable completion results by name.
+def with_args_backwards(x_2)
+  x_1 = nil
+  x_ # error: does not exist
+  # ^ completion: x_1, x_2
+end
+
+def with_kwargs(x_1, x_2:)
+  x_0 = nil
+  x_ # error: does not exist
+  # ^ completion: x_0, x_1, x_2
+end
+
+# We don't look at which locals are actually in scope (and have no plans to)
+def before_after
+  x_1 = nil
+
+  x_ # error: does not exist
+  # ^ completion: x_1, x_2
+
+  x_2 = 1
+end
+
+# We don't yet complete locals when the position is already a valid local
```

Comment:
Might be a good call. I dislike it when I type something perfectly valid and autocomplete has other ideas.

Reply:
I'll make a note to add this to the polish items.

I think it's worth noting that this nearly never comes up in practice, because VS Code keeps the list of completion items open as you're typing and doesn't always issue a new completion request. So what happens is something like this:

```
xyz = 1
xyzabc = 2

# As someone is typing:
x    # completion: xyz, xyzabc
xy   # completion: xyz, xyzabc
xyz # completion: xyz, xyzabc  (still shows both, because VS Code keeps the old results around and filters it itself)

# If someone isn't typing but just asks for completion results:
xyz # completion: (nothing)
```